zerver/lib/actions.py

```diff
@@ -1755,6 +1755,9 @@ def bulk_remove_subscriptions(users, streams):
         if stream.realm.is_zephyr_mirror_realm and not stream.invite_only:
             continue
 
+        if stream.invite_only and stream.num_subscribers() == 0:
+            do_deactivate_stream(stream)
+
```

Comment:
I think this should interact with the  `new_vacant_streams` list; I think basically the bug here is that the subset of streams that are vacated that were invite-only should be deactivated rather than just having "vacate" events sent.

You'll want to add an EventsRegisterTest test for this case, basically similar to `do_test_subscribe_events`.  The testing section at the end of http://zulip.readthedocs.io/en/latest/events-system.html can teach you how this works :).